storage/Utils/Logger.cc

```diff
@@ -89,13 +90,31 @@ namespace storage
     class LogfileLogger : public Logger
     {
     public:
+	LogfileLogger(const std::string& filename, int permissions = DEFAULT_PERMISSIONS);
 
 	virtual void write(LogLevel log_level, const std::string& component, const std::string& file,
 			   int line, const std::string& function, const std::string& content) override;
 
+    private:
+
+	static const int DEFAULT_PERMISSIONS;
+
+	const std::string filename;
+	const int permissions;
     };
 
 
+    const int LogfileLogger::DEFAULT_PERMISSIONS = 0640;
+
+
+    LogfileLogger::LogfileLogger(const string& filename, int permissions) :
+	filename(filename),
+	permissions(permissions)
+    {
+	create_file_if_not_existing(filename, permissions);
```

Comment:
The audit suggests to use an open()/fdopen() sequence.

Reply:
The create_file_if_not_existing function could be removed. And fopen does likely nothing else than open/fdopen anyway.

Why the security team suggests it I cannot say.